app/scripts/controllers/modals/editModal.js

```diff
@@ -0,0 +1,123 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .controller('EditModalController', function($scope, APIService, DataService, QuotaService, LimitRangesService, $uibModalInstance, $filter) {
+
+    var limitRangesVersion = APIService.getPreferredVersion('limitranges');
+    var resourceQuotasVersion = APIService.getPreferredVersion('resourcequotas');
+    var appliedClusterResourceQuotasVersion = APIService.getPreferredVersion('appliedclusterresourcequotas');
+
+    var amountAndUnit = $filter('amountAndUnit');
+    var usageWithUnits = $filter('usageWithUnits');
+    var usageValue = $filter('usageValue');
+
+    var allocatedAmountAndUnit = amountAndUnit($scope.currentCapacity);
+
+    $scope.projectName = JSON.parse($scope.pvc).metadata.namespace;
+    $scope.claim = {}
+    $scope.claim.capacity = Number(allocatedAmountAndUnit[0]);
+    $scope.claim.unit = allocatedAmountAndUnit[1];
+    $scope.disableButton = true;
+
+    $scope.currentCapacityUnits = angular.copy($scope.claim);
+
+    $scope.units = [{
+      value: "Mi",
+      label: "MiB"
+    }, {
+      value: "Gi",
+      label: "GiB"
+    }, {
+      value: "Ti",
+      label: "TiB"
+    }, {
+      value: "M",
+      label: "MB"
+    }, {
+      value: "G",
+      label: "GB"
+    }, {
+      value: "T",
+      label: "TB"
+    }];
+
+    $scope.expand = function() {
+      $scope.updatedCapacity = $scope.claim.capacity + $scope.claim.unit;
+      $uibModalInstance.close($scope.updatedCapacity);
+    };
+
+    $scope.cancel = function() {
+      $uibModalInstance.dismiss('cancel');
+    };
+
+    $scope.groupUnits = function(unit) {
+      switch (unit.value) {
+        case 'Mi':
+        case 'Gi':
+        case 'Ti':
+          return 'Binary Units';
+        case 'M':
+        case 'G':
+        case 'T':
+          return 'Decimal Units';
+      }
+      return '';
+    };
+
+    var validateQuota = function() {
+      var willExceedStorage = QuotaService.willRequestExceedQuota($scope.quotas, $scope.clusterQuotas, 'requests.storage', (($scope.claim.capacity -$scope.currentCapacityUnits.capacity) + $scope.claim.unit));
+      $scope.expandPersistentVolumeClaimForm.capacity.$setValidity('willExceedStorage', !willExceedStorage);
+    };
+
+    var validateLimitRange = function() {
```

Comment:
This looks like it's copied from oscPersistentVolumeClaim.js. Is there a way we can reuse the code without the copy/paste?